pkg/utils/utils.go

```diff
@@ -70,3 +72,39 @@ func DecompressGZIP(compressedData []byte) (decompressedData []byte, err error)
 
 	return decompressedData, nil
 }
+
+// TODO: check if it's worth trying to compress the data prior to compressing it (e.g: using entropy algorithms)
+// This would save some CPU usage client side
+// TODO: have telemetry data type be passed in as a parameter to further check if we should compress data and which algorithm to use (e.g: deflate or gzip)
+func CompressWhenNeeded(data []byte) (resultData []byte, compression *string, err error) {
+	// 'compression' is inserted as a sql.NullString, hence it is returned as a nullable string
+	var validStr string = "gzip"
+
+	// check whether it's worth compressing
+	if len(data) <= 80 {
```

Comment:
Probably a good idea to define this min size before bothering to compress as a constant, just before this routine, and then reference that in the code, e.g. COMPRESSION_DATA_MIN_SIZE or something like that.